server/src/acccount-manager/user.service.ts

```diff
@@ -12,12 +11,11 @@ const { BCRYPT_WORK_FACTOR = '10' } = process.env;
 export class UsersService {
   constructor(@InjectRepository(User) private usersRepository: Repository<User>) {}
 
-  async create(createUserDto: CreateUserInternal): Promise<ReturnUserDto> {
+  async create(createUserDto: CreateUserInternal): Promise<User> {
     try {
       const hashedPw = await bcrypt.hash(createUserDto.password, parseInt(BCRYPT_WORK_FACTOR));
       createUserDto.password = hashedPw;
       const user = await this.usersRepository.save(createUserDto);
-      delete user.password;
```

Comment:
I've verified locally that `password` is not included in the returned payload even with this line removed.  I'm not 100% clear on how this is happening, so describing here what I think is going on.  In `UsersService.create` method, we've replaced `ReturnUserDto` as the output type with `User`, but in the controller, we use annotation `@MapTo(ReturnUserDto)`, which I'm assuming is doing the sanitizing.   Since `password` is not one of the properties of `ReturnUserDto`, what does the `@Expose()` annotation do and why do we use it?

Reply:
Yes, in a way. changing the return type actually has no bearing on the returned data (thanks for Javascript). Thankfully, `class-transformer` is very good at this. The class-transformer interceptor is getting set to
```typescript
 return plainToInstance(this.dto, data, {
          excludeExtraneousValues: true,
          strategy: 'excludeAll',
        });
```

The `excludeAll` strategy means that any `ReturnDto` property not decorated with `Expose` will be removed from the response.

Reply:
I went with `excludeAll` so that we have to explicitly list all the data we want to expose, but we can change it if it becomes to cumbersome.